src/main/java/com/puppycrawl/tools/checkstyle/xpath/ElementNode.java

```diff
@@ -40,6 +40,9 @@
     /** String literal for text attribute. */
     private static final String TEXT_ATTRIBUTE_NAME = "text";
 
+    /** Constant for optimization. */
+    private static final AbstractNode[] EMPTY_ABSTRACT_NODE_ARRAY = new AbstractNode[0];
```

Comment:
If you don't want to move this to `CommonUtil`, why don't we move it to `AbstractNode` so all classes can use it?

Reply:
yes, this is not very clear point.
CommonUtil is used for common empty arrays of jdk standard types.
BUT new empty arrays are not JDK, they are custom to xpath. Placing them to AbstractNode as public member is not very good, as it is not clear what for and how to use it. Placing them to AbstractNode as protected ..... kind of ok, but i personally dislike protected, even in this case looks like no harm possible. So I decided to keep them private close to place where "optimization" took place, ones some another good practice appear - it will be easy to refactor.